modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/preloader/GridDhtPartitionsExchangeFuture.java

```diff
@@ -1628,6 +1660,100 @@ private void changeWalModeIfNeeded() {
         return null;
     }
 
+    /**
+     * @throws IgniteCheckedException If failed.
+     */
+    private void waitRecovery() throws IgniteCheckedException {
```

Comment:
The code in this method is very much similar to the waitPartitionRelease(). Can we avoid the duplication?